Code check

GiniSt19 opened the pull request.

GiniSt19 committed:
feat: initial plugin setup

basic classes and block structure

GiniSt19 committed:
feat: basic block, Settings Page & Shortcode

- #3 basic block &
- # 11 basic Settings Page
- #13 Shortcode

GiniSt19 committed:
feat: extend block with additional options

GiniSt19 committed:
feat: extend block with additional options

GiniSt19 committed:
feat: extend block with additional options

GiniSt19 committed:
feat: gallery expanded

Gallery functionality based on theme handlers
#6

GiniSt19 committed:
feat: block options, translations, API prepared

GiniSt19 committed:
feat: API connection and block adjustments

GiniSt19 committed:
update: access rights & block options

GiniSt19 committed:
update: Rest API & block options

GiniSt19 committed:
update: block options

GiniSt19 committed:
update: block options & translations

GiniSt19 committed:
feat: preview links & code improvement

GiniSt19 committed:
update: preview link removed & new translations

GiniSt19 committed:
fix: Shortcode fixes, readme revised

lukasniebler commented on .idea/.gitignore:
das ist ein IdE-File und sollte gelöscht werden

lukasniebler commented on .idea/modules.xml:
PHP Storm File sollte gelöscht werden

lukasniebler commented on .idea/php.xml:
PHP Storm file sollte gelöscht werden

lukasniebler commented on .idea/vcs.xml:
PHP Storm file sollte gelöscht werden

lukasniebler commented on composer.json:
Der PSR-4 Autoloader liegt bei dir in der Main-Plugin-datei. D.h. du kannst composer.json löschen, weil wir es nicht zum Autoloading benutzen. 

Das Autoloading wird bei deinem Plugin in der Haupt-Plugindatei in Zeile 
```php
spl_autoload_register(function ($class) {
    $prefix = __NAMESPACE__ . '\\';
    $baseDir = __DIR__ . '/includes/';

    $len = strlen($prefix);
    if (strncmp($prefix, $class, $len) !== 0) {
        return;
    }

    $relativeClass = substr($class, $len);
    $file = $baseDir . str_replace('\\', '/', $relativeClass) . '.php';

    if (file_exists($file)) {
        require $file;
    }
});
```

Abgehandelt => Deshalb brauchen wir gar kein composer.json

lukasniebler commented on includes/Main.php:
```diff
@@ -0,0 +1,67 @@
+//        if (is_admin()) {
+//            Admin\Settings::register();
+//        }
+//    }
```
Auskommentierte Funktionen löschen wir am besten gleich raus. D.h. in diesem File Zeile 29, 37-42

lukasniebler commented on includes/Main.php:
```diff
@@ -0,0 +1,67 @@
+     * Currently provides a dummy folder list for use in the block editor.
+     * Replace or extend with real API routes later.
+     */
+    private function restAPI(): void
```
Die Funktion würde ich umbenennen in ```registerRestRoute()``` – Dadurch darfst du dann Zeile 54 bis 59 löschen, da der Funktionsname selbsterklärend ist.

Zeile 32 musst du dann noch anpassen zu ```$this->registerRestRoute();```

lukasniebler commented on includes/Main.php:
```diff
@@ -0,0 +1,67 @@
+        Rest::register();
+    }
+
+
```
Auf die Formatierung achten. Manchmal sind etwas viele Leerzeilen enthalten.

lukasniebler commented on includes/Shortcode.php:
```diff
@@ -0,0 +1,193 @@
+        $resourceId = API::resolveResourceId($shareId);
+        if (!$resourceId) {
+            return '<p>' . esc_html__('Invalid FAUbox public link.', 'rrze-faubox') . '</p>';
+        }
```
Zeile 26 bis 46 lagerst du am besten in eine separate Funktion aus.

z.B.:
```
public static function validateShareLink(string $shareLink): void | string
{
// Funktionskörper
}
```

Das macht deine Render-Funktion kürzer und leichter durchschaubar.

lukasniebler commented on includes/Shortcode.php:
```diff
@@ -0,0 +1,193 @@
+        if (empty($selectedFolders)) {
+            $allItems = API::fetchRoot($resourceId, $shareId) ?? [];
+        }
+
```
3 5 6 7 und vielleicht auch 8 könntest du auch in separate kleinere Funktionen auslagern. Das erhöht auf Dauer Wartbarkeit und Lesbarkeit. 

Durch das auslagern und sprechende Funktionsnamen kannst du dann auch die Kommentare weglassen.

lukasniebler commented on readme.txt:
```diff
@@ -0,0 +1,87 @@
+=== RRZE FAUbox ===
+Contributors: rrze
+Tags: files, cloud, fau, faubox, directory, list, table, preview
+Tested up to: 6.5
```
6.9

lukasniebler commented on src/block/edit.tsx:
```diff
@@ -0,0 +1,274 @@
+                    </div>
+
+
+                </Placeholder>
```
Aktuell muss man als User glaube ich nach der Eingabe der Share-URL in die InspectorControls wechseln um die Einrichtung abzuschließen. Eigentlich wäre es schön, wenn man als User auch die Ordner-Auswahl im Placeholder noch bestätigen kann, bevor der Placeholder ausgeblendet wird.

lukasniebler commented on src/block/index.tsx:
```diff
@@ -0,0 +1,12 @@
+import './editor.scss'; // 🟢 Styles für Editor laden
```
Kommentare können entfernt werden

lukasniebler commented on rrze-faubox.php:
```diff
@@ -0,0 +1,80 @@
+        ['wp-blocks', 'wp-element', 'wp-i18n', 'wp-components', 'wp-block-editor', 'wp-api-fetch'],
+        '1.0.0',
+        true
+    );
```
Ich weiß noch nicht ob ich von der Blockregistrierung hier begeistert bin. Den Block via wp_register_script zu registrieren ist eigentlich sehr altbacken. Vgl. https://developer.wordpress.org/block-editor/getting-started/fundamentals/registration-of-a-block/

lukasniebler commented on includes/API.php:
```diff
@@ -0,0 +1,281 @@
+            return $cached;
+        }
+
+
```
Formatierung noch anpassen. Oft sind 2-3 Leerzeilen im Code.

lukasniebler commented on includes/Admin/Rest.php:
```diff
@@ -0,0 +1,206 @@
+        $resourceId = API::resolveResourceId($shareId);
+        if (!$resourceId) {
+            return new WP_REST_Response([], 200);
+        }
```
Die Funktion ist sehr lang. Kann man sie in kleinere Funktionen aufsplittern?

validateShareLink
etc.

Je übersichtlicher, desto besser.

lukasniebler commented on includes/Blocks/BlockRender.php:
```diff
@@ -0,0 +1,29 @@
+     */
+    public static function output(array $attributes = []): string
+    {
+        // Passes block attributes directly to the shortcode renderer
```
Z24 löschen

lukasniebler commented on includes/Renderer.php:
```diff
@@ -0,0 +1,164 @@
+//    private static function getPreviewIcon(): string
+//    {
+//        $path = plugin_dir_path(__DIR__, 2) . 'assets/svg/image_arrow_up.svg';
+//
```
51+ auskommentierte Funktionen am besten nicht im Produktionscode

lukasniebler commented on includes/Renderer.php:
```diff
@@ -0,0 +1,164 @@
+
+        $columns = $atts['show'] ?? ['name'];
+
+        // 🟩 Mapping: internal column → translated label
```
ggf. kann man das mapping in eine gesonderte Funktion auslagern.

lukasniebler reviewed (changes requested):
Alles in allem gut, aber noch Verbesserungspotential an der ein oder anderen Stelle (sonst wäre es ja langweilig) ;-)

GiniSt19 committed:
Delete .idea/.gitignore

GiniSt19 committed:
Delete .idea/modules.xml

GiniSt19 committed:
Delete .idea/php.xml

GiniSt19 committed:
Delete .idea/vcs.xml

GiniSt19 committed:
fix: code fixes Pull Request

GiniSt19 committed:
Merge branch 'dev' of https://github.com/RRZE-Webteam/rrze-faubox into dev

GiniSt19 committed:
fix: code fixes Pull Request

GiniSt19 committed:
fix: code fixes

GiniSt19 committed:
feat: Add debug helper and system requirements check

GiniSt19 committed:
refactor: Add FAUbox API client and settings

Introduce server-side FAUbox integration: add API Client, FileService and REST controller plus an Admin Settings page to store credentials. Refactor block and renderer to use a folder path and orderby attribute (replace public share-link flow), move Renderer to Frontend, and update Blocks registration and rendering.

GiniSt19 committed:
feat: Add file size/date display and UI options

Extend block attributes and editor controls to support file size and modification date. Updated block metadata (show and orderby enums include "size" and "modified") and added corresponding checkboxes in the editor, a placeholder select option, a text control to overwrite folder name, and consolidated file-type filters and sorting controls.

GiniSt19 committed:
feat: block layout adjustments

GiniSt19 committed:
feat: Add heading_level option to FAUbox block

Introduce a new heading_level attribute (enum: h2,h3,h4,h5; default: h3) to the FAUbox block. Add a SelectControl in the block editor to choose heading level and wire it into attributes; update block metadata, compiled assets and asset version, and refresh translations and server/frontend renderer files accordingly.

GiniSt19 committed:
fix: missing .mo file

GiniSt19 committed:
feat: Add subfolder endpoint and parentPath support

Introduce FileService::getSubFolders to fetch WebDAV subfolders. Extend the REST /folders route to accept an optional 'path' argument and update RestController::getFolders to return subfolders when a path is provided. Add a new 'parentPath' attribute to block.json and enhance src/block/edit.tsx to support selecting a parent folder and loading/displaying its subfolders (new state for subFolders/subLoading, API fetch on parentPath change, and updated SelectControls). Also includes minor formatting/indent fixes in RestController.

GiniSt19 committed:
Build: add parentPath and refresh assets

GiniSt19 committed:
Update German translations and add subfolder

Refresh German (de_DE) localization: bump POT and PO revision dates, update .po/.mo and JSON locale files, and refine the "FAUbox Main Folder" message to include the optional root-folder note. Added new block strings: "Subfolder" and "— All files in this folder —" and updated source reference locations in the translation entries.

GiniSt19 committed:
update: translations

GiniSt19 committed:
feat: Rebuild assets and improve block editor

Recompiled block/build assets and updated editor code. Added/updated built files (build/block/index.js, index.css, index-rtl.css, index.js.map, build/admin/admin.css) and bumped asset version. Improve block editor logic (src/block/edit.tsx): more robust folder/subfolder loading, handle no_folder_configured error, clearer UI and filetype handling. Also updated PHP integration and API classes (Client, FileService, Admin settings, BlockRegistration, RestController), translations, package.json and plugin entry; added source admin SCSS.

GiniSt19 committed:
feat: Use folder tree selector in FAUbox block

Add FolderTree and FolderTreeNode React components and update the block editor to use a collapsible folder tree for selecting FAUbox folders (replacing the previous parentPath/subfolder selects). Remove the parentPath attribute from block.json and update src/block/edit.tsx to integrate the new tree UI and loading logic. Regenerate build artifacts: minified CSS, updated index.js bundle and asset version, and remove the source map. Also include small related updates to the PHP client and frontend renderer.
#15
